trio/_core/_io_windows.py

```diff
@@ -632,9 +667,12 @@ def _refresh_afd(self, base_handle):
                     # Do this last, because it could raise.
                     wake_all(waiters, exc)
                     return
-            op = AFDPollOp(lpOverlapped, poll_info, waiters)
+            op = AFDPollOp(lpOverlapped, poll_info, waiters, afd_group)
             waiters.current_op = op
             self._afd_ops[lpOverlapped] = op
+            afd_group.size += 1
+            if afd_group.size < MAX_AFD_GROUP_SIZE:
+                self._vacant_afd_groups.add(afd_group)
```

Comment:
Hmm, I think there's a real bug here, not just a style nitpick like my other comments :-). If the `DeviceIoControl` call fails and the function exits early, then we can "lose" the `afd_group` -- it still has vacancies, but we won't put it back in `self._vacant_afd_groups`. I think the clearest way would be to leave the group in `_vacant_afd_groups` until we're ready to increment `.size`.

It's annoying there's no equivalent of `pop` that *doesn't* remove the item. I guess the choices are `afd_group = next(iter(self._vacant_afd_groups))` or `afd_group = self._vacant_afd_groups.pop(); self._vacant_afd_groups.add(afd_group)`.